src/transforms/groupTransforms.ts

```diff
@@ -7,14 +7,29 @@ import {
 import { getLanguageMetadata } from "../utils/languageHelpers";
 import { ObjectDetails, ObjectKind } from "../models/modelDetails";
 import { transformProperty } from "./objectTransforms";
+import { OptionsBag } from "../utils/optionsBag";
 
-export function transformGroups(codeModel: CodeModel) {
-  return (codeModel.schemas.groups || []).map(transformGroup);
+export function transformGroups(codeModel: CodeModel, optionsBag: OptionsBag) {
+  codeModel.schemas.groups = codeModel.schemas.groups || [];
+  const objectDetails = [];
+  for (const grp of codeModel.schemas.groups) {
+    objectDetails.push(transformGroup(grp, optionsBag));
+  }
+  return objectDetails;
+  // return (codeModel.schemas.groups || []).map(transformGroup);
```

Comment:
remove this commented line?

Reply:
I'm curious about this change, would it be the same as 

```typescript
return (codeModel.schemas.groups ?? []).map(group => transformGroup(group, optionsBag));
```

I like more the map approach as it is easier to read and has less surface for bugs

Reply:
Done